src/coreclr/classlibnative/bcltype/system.cpp

```diff
@@ -273,19 +277,22 @@ void SystemNative::GenericFailFast(STRINGREF refMesgString, EXCEPTIONREF refExce
         {
             // Truncate the message to what will fit in the static buffer.
             cchMessage = FAIL_FAST_STATIC_BUFFER_LENGTH - 1;
-            pszMessage = g_szFailFastBuffer;
+            pszMessage = InterlockedExchangeT(&g_pFailFastBuffer, NULL);
         }
     }
 
     if (cchMessage > 0)
         memcpyNoGCRefs(pszMessage, gc.refMesgString->GetBuffer(), cchMessage * sizeof(WCHAR));
-    pszMessage[cchMessage] = W('\0');
 
-    if (cchMessage == 0) {
+    if (pszMessage == NULL) {
+        WszOutputDebugString(W("CLR: Managed code called FailFast, but there is not enough memory to print the supplied message.\r\n"));
+    }
+    else if (cchMessage == 0) {
         WszOutputDebugString(W("CLR: Managed code called FailFast without specifying a reason.\r\n"));
     }
     else {
         WszOutputDebugString(W("CLR: Managed code called FailFast, saying \""));
```

Comment:
Not your change, but to be consistent with the other CLR termination messages, this would be `...called FailFast. "));`. They all suffix the message after a period.

Reply:
I suggest to keep it on one line (ie, space instead of `\r\n`). That is the existing pattern and it works well with logs and concurrent console output where the two lines can otherwise get separated.